client/asset/eth/nodeclient_harness_test.go

```diff
@@ -143,69 +139,77 @@ func newRedeem(secret, secretHash [32]byte) *asset.Redemption {
-	}()
-	timesUp := time.After(timeLimit)
-out:
+	hash := tx.Hash()
+	// Waiting as much as five blocks.
+	timesUp := time.After(5 * secPerBlock)
 	for {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
-		case <-timesUp:
-			return errors.New("timed out")
+			return nil, ctx.Err()
 		case <-time.After(time.Second):
-			txsa, err := nc.pendingTransactions()
-			if err != nil {
-				return fmt.Errorf("initiator pendingTransactions error: %v", err)
-			}
-			txsb, err := participantEthClient.pendingTransactions()
+			receipt, err := nc.transactionReceipt(ctx, hash)
 			if err != nil {
-				return fmt.Errorf("participant pendingTransactions error: %v", err)
-			}
-			if len(txsa)+len(txsb) == 0 {
-				break out
+				if errors.Is(err, asset.CoinNotFoundError) {
+					continue
+				}
+				return nil, err
 			}
+			spew.Dump(receipt)
+			return receipt, nil
+		case <-timesUp:
+			return nil, errors.New("wait for receipt timed out, txn might be missing due to reorg, check the following tx for a bad nonce or low gas cap")
+			spew.Dump(tx)
 		}
 	}
-	if waitTimeLimit {
-		<-timesUp
-	}
-	return nil
 }
 
-// waitForMinedTestnet will multiply the time limit by testnetSecPerBlock.
-func waitForMinedTestnet(t *testing.T, nc *nodeClient, timeLimit time.Duration, waitTimeLimit bool) error {
-	t.Helper()
-	timesUp := time.After(timeLimit * testnetSecPerBlock)
-out:
-	for {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case <-timesUp:
-			return errors.New("timed out")
-		case <-time.After(time.Second * testnetSecPerBlock):
-			txs, err := nc.pendingTransactions()
+// waitForMinedFn will multiply the time limit by testnetSecPerBlock for
+// testnet and mine blocks when on simnet.
+func waitForMinedFn() func(*testing.T, *nodeClient, int, bool) error {
+	return func(t *testing.T, nc *nodeClient, nBlock int, waitTimeLimit bool) error {
+		t.Helper()
+		if !isTestnet {
+			err := exec.Command("geth", "--datadir="+alphaNodeDir, "attach", "--exec", "miner.start()").Run()
 			if err != nil {
 				return err
 			}
-			if len(txs) == 0 {
-				break out
+			defer func() {
+				_ = exec.Command("geth", "--datadir="+alphaNodeDir, "attach", "--exec", "miner.stop()").Run()
+			}()
+		}
+		timesUp := time.After(time.Duration(nBlock) * secPerBlock)
+	out:
+		for {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case <-timesUp:
+				return errors.New("timed out")
+			case <-time.After(time.Second):
+				txsa, err := nc.pendingTransactions()
+				if err != nil {
+					return fmt.Errorf("initiator pendingTransactions error: %v", err)
+				}
+				txsb, err := participantEthClient.pendingTransactions()
+				if err != nil {
+					return fmt.Errorf("participant pendingTransactions error: %v", err)
+				}
+				if len(txsa)+len(txsb) == 0 {
+					break out
+				}
 			}
 		}
+		if waitTimeLimit {
+			<-timesUp
+		}
+		// Sleeping just in case the node balance needs to catch up.
+		time.Sleep(time.Second)
```

Comment:
Had a failure for the balance on testInitiate which makes me a little nervous. That was before this sleep. Not sure if there's another reason. Leaving the logs here.

Failed balance check:
[out.txt](https://github.com/decred/dcrdex/files/9038460/out.txt)